cli: passing wrong peer Id to glustercli peer remove command cause panic

#### Logs
```logs
[okumar@localhost glusterd2]$ ./build/glustercli --endpoints=http://192.168.122.253:24007 peer remove thisIsAnInvalidPeerId
Peer remove failed

Response headers:
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x38 pc=0x711a19]

goroutine 1 [running]:
github.com/gluster/glusterd2/glustercli/cmd.failure(0x7fdb01, 0x12, 0x845580, 0xc42009f070, 0x1)
	/home/okumar/GlusterProjects/GlusterD2/src/github.com/gluster/glusterd2/glustercli/cmd/common.go:63 +0x159
github.com/gluster/glusterd2/glustercli/cmd.glob..func24(0xa164e0, 0xc42012a880, 0x1, 0x2)
	/home/okumar/GlusterProjects/GlusterD2/src/github.com/gluster/glusterd2/glustercli/cmd/peer.go:98 +0xf3
github.com/gluster/glusterd2/vendor/github.com/spf13/cobra.(*Command).execute(0xa164e0, 0xc42012a820, 0x2, 0x2, 0xa164e0, 0xc42012a820)
	/home/okumar/GlusterProjects/GlusterD2/src/github.com/gluster/glusterd2/vendor/github.com/spf13/cobra/command.go:702 +0x2c6
github.com/gluster/glusterd2/vendor/github.com/spf13/cobra.(*Command).ExecuteC(0xa16b40, 0xc42006ff30, 0x5, 0xc4200ac130)
	/home/okumar/GlusterProjects/GlusterD2/src/github.com/gluster/glusterd2/vendor/github.com/spf13/cobra/command.go:783 +0x2e4
github.com/gluster/glusterd2/vendor/github.com/spf13/cobra.(*Command).Execute(0xa16b40, 0x7903c0, 0xc4200a1cb0)
	/home/okumar/GlusterProjects/GlusterD2/src/github.com/gluster/glusterd2/vendor/github.com/spf13/cobra/command.go:736 +0x2b
main.main()
	/home/okumar/GlusterProjects/GlusterD2/src/github.com/gluster/glusterd2/glustercli/main.go:14 +0x36

```

--- a/glustercli/cmd/common.go
+++ b/glustercli/cmd/common.go
@@ -56,9 +56,14 @@ func failure(msg string, err error, errcode int) {
 
 	w.WriteString(msg + "\n")
 
-	if err != nil {
-		resp := client.LastErrorResponse()
+	resp := client.LastErrorResponse()
 
+	if resp == nil && err != nil {
+		fmt.Fprintln(w, err)
+		os.Exit(errcode)
+	}
+
+	if err != nil {
 		w.WriteString("\nResponse headers:\n")
 		for k, v := range resp.Header {
 			if strings.HasSuffix(k, "-Id") {

